### PR TITLE
SALTO-6736: Add Elapsed Time to Salesforce Metadata Deploy Progress Reporter

### PR DESCRIPTION
**Pull request by tamtamirr**
SALTO-6736: Add Elapsed Time to Salesforce Metadata Deploy Progress Reporter

---

Example for queued deployment: 
Metadata: Waiting on another deploy or automated process to finish in Salesforce. Elapsed Time: 3.344 seconds. View deployments [in Salesforce](https://salto-1a-dev-ed.develop.my.salesforce.com/lightning/setup/DeployStatus/home)

---
_Release Notes_: 
_Salesforce Adapter_:
- Improved the Metadata Deploy progress to include the elapsed time since the deployment was created.

---
_User Notifications_: 
_None_

### CODE REVIEW

File: packages/salesforce-adapter/src/adapter_creator.ts [hunk 1]
```diff
@@ -253,10 +253,13 @@ export const createDeployProgressReporter = async (
...
     let dataProgress: string | undefined
     if (deployResult) {
+      const startTime = new Date(deployResult.createdDate).getTime()
+      const currentTime = new Date().getTime()
+      const elapsedTimeMinutes = ((currentTime - startTime) / (1000 * 60)).toFixed(1)
       metadataProgress =
         deployResult.status === METADATA_DEPLOY_PENDING_STATUS
-          ? 'Metadata: Waiting on another deploy or automated process to finish in Salesforce'
-          : `${deployResult.numberComponentsDeployed}/${deployResult.numberComponentsTotal} Metadata Components, ${deployResult.numberTestsCompleted}/${deployResult.numberTestsTotal} Tests.${linkToSalesforce(deployResult)}`
+          ? `Metadata: Waiting on another deploy or automated process to finish in Salesforce. Elapsed Time: ${elapsedTimeMinutes} minutes`
```
Comment (yelly) [inline packages/salesforce-adapter/src/adapter_creator.ts:261]: Maybe we can use something like https://www.npmjs.com/package/humanize-duration here to make it look less silly? ("0.1 minutes" doesn't really parse for humans)

File: packages/salesforce-adapter/src/adapter_creator.ts [hunk 2]
```diff
@@ -253,10 +254,13 @@ export const createDeployProgressReporter = async (
...
     let dataProgress: string | undefined
     if (deployResult) {
+      const startTime = new Date(deployResult.createdDate).getTime()
+      const currentTime = new Date().getTime()
+      const elapsedTimeMinutes = humanizeDuration(currentTime - startTime)
       metadataProgress =
         deployResult.status === METADATA_DEPLOY_PENDING_STATUS
-          ? 'Metadata: Waiting on another deploy or automated process to finish in Salesforce'
-          : `${deployResult.numberComponentsDeployed}/${deployResult.numberComponentsTotal} Metadata Components, ${deployResult.numberTestsCompleted}/${deployResult.numberTestsTotal} Tests.${linkToSalesforce(deployResult)}`
+          ? `Metadata: Waiting on another deploy or automated process to finish in Salesforce. Elapsed Time: ${elapsedTimeMinutes}`
```
Comment (adamro) [inline packages/salesforce-adapter/src/adapter_creator.ts:262]: Can we also add a link to the deploy queue in this case please?

File: packages/salesforce-adapter/src/adapter_creator.ts [hunk 3]
```diff
@@ -249,14 +250,24 @@ export const createDeployProgressReporter = async (
     return ` View ${deploymentOrValidation} status [in Salesforce](${deploymentUrl})`
   }
 
+  const linkToSalesforceDeploymentsPage = (): string => {
+    if (!baseUrl) {
+      return ''
+    }
+    return ` View deployments [in Salesforce](${baseUrl}lightning/setup/DeployStatus/home)`
+  }
```
Comment (yelly) [inline packages/salesforce-adapter/src/adapter_creator.ts:258]: Nit: why is this a function?
```suggestion
  const salesforceDeploymentsPageLinkText = baseUrl ? ` View deployments [in Salesforce](${baseUrl}lightning/setup/DeployStatus/home)` : ''
```

File: packages/salesforce-adapter/src/adapter_creator.ts [hunk 4]
```diff
@@ -249,14 +250,24 @@ export const createDeployProgressReporter = async (
...
     if (deployResult) {
+      const startTime = new Date(deployResult.createdDate).getTime()
+      const currentTime = new Date().getTime()
+      const elapsedTimeMinutes = humanizeDuration(currentTime - startTime)
       metadataProgress =
         deployResult.status === METADATA_DEPLOY_PENDING_STATUS
-          ? 'Metadata: Waiting on another deploy or automated process to finish in Salesforce'
-          : `${deployResult.numberComponentsDeployed}/${deployResult.numberComponentsTotal} Metadata Components, ${deployResult.numberTestsCompleted}/${deployResult.numberTestsTotal} Tests.${linkToSalesforce(deployResult)}`
+          ? `Metadata: Waiting on another deploy or automated process to finish in Salesforce. Elapsed Time: ${elapsedTimeMinutes}.${linkToSalesforceDeploymentsPage()}`
+          : `${deployResult.numberComponentsDeployed}/${deployResult.numberComponentsTotal} Metadata Components, ${deployResult.numberTestsCompleted}/${deployResult.numberTestsTotal} Tests. Elapsed Time: ${elapsedTimeMinutes}.${linkToSalesforceDeployment(deployResult)}`
```
Comment (yelly) [inline packages/salesforce-adapter/src/adapter_creator.ts:270]: Is this not covered by any test?

**Pull request merged**